satellite/peer.go

```diff
@@ -491,6 +491,21 @@ func ignoreCancel(err error) error {
 	return err
 }
 
+func (peer *Peer) restartLoop(ctx context.Context, name string, service func(context.Context) error) error {
```

Comment:
This seems like a really dangerous thing to do. If something panics the memory might be corrupted.

Reply:
net/http does it?

Reply:
really?

Reply:
but sure, the panic handling isn't the most important thing about this. given that net/http's Serve does it, do you still want me to take it out?

Reply:
Yeah, looks like it. :/

Reply:
I'm usually in favor of crash hard and early, because tracing down the actual bug and damage after the fact can be much more difficult. But I understand if we want to keep satellite up and running no matter the cost.

Reply:
yeah i think we do